pkg/apis/kai/v1/pod_group_controller/pod_group_controller.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+// +kubebuilder:object:generate:=true
+package pod_group_controller
+
+import (
+	"github.com/NVIDIA/KAI-scheduler/pkg/apis/kai/v1/common"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/ptr"
+)
+
+const (
+	imageName = "pod-group-controller"
+)
+
+type PodGroupController struct {
+	// Enabled defines whether the pod-group-controller should be deployed
+	// +kubebuilder:validation:Optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Image is the configuration of the pod-group-controller image
+	// +kubebuilder:validation:Optional
+	Image *common.Image `json:"image,omitempty"`
```

Comment:
An idea - maybe Image, Resources, and others could be a part of "ControllerCommon" struct - all other controllers will use this struct, and will be able to use the same logic to generate an initial deployment from, and only modify specific things if needed (most will probably only need to override the image and args)

Reply:
So a typical operand will be something like:
```Go
deployment := config.podGroupController.controllerCommon.GetDeployment()
deployment.Spec.Template.Spec.Containers[0].Args = getArgs(config.podGroupController.args)
```

Something like that

Reply:
its possible to just create this struct and embed it without name so the usage will be the same but it will save a lot of duplications in definition. :+1: